zospy/analyses/mtf.py

```diff
@@ -237,3 +237,125 @@ def fft_through_focus_mtf_fromcfg(
     )
 
     return analysis.complete(oncomplete, result)
+
+
+def huygens_mtf(
+    oss: OpticStudioSystem,
+    pupilsampling: str | int = "32x32",
+    imagesampling: str | int = "32x32",
+    imagedelta: float = 0.0,
+    wavelength: str | int = "All",
+    field: str | int = "All",
+    mtftype: constants.Analysis.Settings.Mtf.HuygensMtfTypes | str = "Modulation",
+    maximumfrequency: float = 150.0,
+    use_polarization: bool = False,
+    use_dashes: bool = False,
+    oncomplete: OnComplete | str = OnComplete.Close,
+) -> AnalysisResult:
+    """Wrapper around the OpticStudio Huygens MTF.
+
+    For an in depth explanation of the parameters, see the Zemax OpticStudio user manual
+
+    Parameters
+    ----------
+    oss: zospy.core.OpticStudioSystem
+        A ZOSPy OpticStudioSystem instance. Should be sequential.
+    pupilsampling: str | int
+        The pupil sampling, either string (e.g. '64x64') or int.
+        The integer will be treated as a ZOSAPI Constants integer.
+    imagesampling: str | int
+        The image sampling, either string (e.g., '64x64') or int.
+        The integer will be treated as a ZOSAPI Constants integer.
+    imagedelta: float
+        The Image Delta, defaults to 0.0.
+    wavelength: str | int
+        The wavelength to use in the MTF. Either 'All' or an integer specifying the wavelength number.
+    field: str | int
+        The field to use in the MTF. Either 'All' or an integer specifying the field number.
+    mtftype: zospy.constants.Analysis.Settings.Mtf.MtfTypes.Modulation
+        The MTF type (e.g. `Modulation`) that is calculated.
+    maximumfrequency: float
+        The maximum frequency at which the MTF is calculated in lp/mm. Defaults to 150.0.
```

Comment:
> lp/mm

Maybe a stupid question, but what is this unit? (I don't use this type of analysis in my own research).

Reply:
I also see OpticStudio allows to change this unit, so I don't think it should be in the docstring.

Reply:
This stands for line pairs per mm. I don't see where OpticStudio allows one to change this unit for the Huygens MTF analysis.

Reply:
The OpticStudio documentation states the frequency is in "MTF Units". These can be adjusted in the System Explorer under Units.

All available units are Cycles / something; are line pairs the same as cycles?

Reply:
I believe they are, but we can label it cycles. You are right - you can change between cycles/mm and cycles/milliradian. I think the method would still work if the system had been configured for cycles/milliradian, but the `maximum_frequency` argument default value should perhaps be a function of the "MTF Units" set in the system - I am not sure if this is possible in Python, actually.

Maybe this is something for another PR - units of cycles/(angular quantity) are somewhat uncommon from what I can tell.

Reply:
For now, I have addressed this by modifying the docstring in https://github.com/MREYE-LUMC/ZOSPy/pull/55/commits/ef9f57c2d82a58a463c9fbe66711de0fa51a9f6c.

Reply:
I think it's best to refer to "MTF units" and mention the available units, along the line of what you did (and then hopefully Ansys won't add more units, because we'll probably forget to add them to this docstring...)

In my opinion, the MTF unit should not be modified from an analysis, because this has system-wide effects. The user should do this explicitly from `oss.SystemData`.